pkg/kp/service.go

```diff
@@ -7,6 +7,12 @@ import (
 	"github.com/square/p2/pkg/pods"
 )
 
+var (
+	// prints the HTTP response on stderr, while exiting 0 if the status code is 200
+	httpStatusCheck  = `if [[ $(curl http://$(hostname):%v/_status -s -o /dev/stderr -w "%%{http_code}") == "200" ]] ; then exit 0 ; else exit 2; fi`
+	httpsStatusCheck = `if [[ $(curl https://$(hostname):%v/_status -s -o /dev/stderr -w "%%{http_code}" --cacert ${SECRETS_PATH}/service2service.ca.pem) == "200" ]] ; then exit 0 ; else exit 2; fi`
```

Comment:
The problem with putting it in the preparer config is that you have to set it to `/data/pods/consul/secrets/service2service.ca.pem`, which is awful for a variety of reasons:

1. writing down the path to secrets
2. writing down the id of a pod...
3. and the pod id you're referencing isn't even your own

You cannot use the preparer's s2s cert because it's consul that ends up executing the curl. They are the same cert yes, but Consul does not have read permissions on that particular copy. I agree that the hardcoding is gross, but this configuration option does not look much better to me.

Also, you won't be affected by this at bootstrap time because neither consul nor the preparer have health checks. http health checks will also be unaffected because they do not attempt to set a `--ca-cert`. You won't actually notice the missing cert at all unless you attempt to schedule a pod manifest that contains an https health check.